### PR TITLE
Quash

### LINKED ISSUE #1803: Quash might be broken
This is another bug reported in my hack and not confirmed in the BE yet. Apparently Quash not only doesn't make the target move last, it makes the target move twice in the same turn. I suspect that it moves the target's action to the end of the move/turn order, but doesn't remove its initial action somehow.

### PR DESCRIPTION
**Pull request by DizzyEggg**
This fixes #1803

It's based on this old PR #1974, but its owner seems to be inactive so I pulled their repo and did the requested changes.

### CODE REVIEW

**Review by ghoulslash — CHANGES_REQUESTED** [review]

File: src/battle_main.c [hunk 1]
```diff
@@ -4736,6 +4739,7 @@ static void TurnValuesCleanUp(bool8 var0)
             gProtectStructs[gActiveBattler].spikyShielded = FALSE;
             gProtectStructs[gActiveBattler].kingsShielded = FALSE;
             gProtectStructs[gActiveBattler].banefulBunkered = FALSE;
+            gProtectStructs[gBattlerTarget].quash = FALSE;
```
[the remark this answers is not in the document]
Comment (DizzyEggg) [reply src/battle_main.c:4742]: also missing from FaintClearSetData?

[the remark this answers is not in the document]
Comment (ghoulslash) [reply src/battle_main.c:4742]: The protect struct fields are all cleared on fainting [here](https://github.com/rh-hideout/pokeemerald-expansion/blob/ae2ff4c1d88693aa06cdd5bc3796f7b7c83d2099/src/battle_main.c#L3167).

[the remark this answers is not in the document]
Comment (DizzyEggg) [reply src/battle_main.c]: Fixed

File: src/battle_main.c [hunk 2]
```diff
@@ -4516,6 +4517,9 @@ s8 GetMovePriority(u32 battlerId, u16 move)
         }
     }
 
+    if (gProtectStructs[battlerId].quash)
+		priority = -8;
```
Comment (AsparagusEduardo) [inline src/battle_main.c:4521]: Spaces instead of tabs

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]

File: src/battle_main.c (hunk 2 above)
Comment (DizzyEggg) [reply src/battle_main.c]: done

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**